packages/config/src/index.ts

```diff
@@ -33,8 +33,14 @@ export interface BlitzConfig extends Record<string, unknown> {
   _meta: {
     packageName: string
   }
+  middleware?: {
+    (req: any, res: any, next: any): Promise<void> | void
+    type?: string
+    config?: {
+      cookiePrefix?: string
+    }
+  }[]
```

Comment:
1. Change `type` to `name` here
2. And allow any attributes on middleware. So probably extract this to an interface that `extends Record<string, any>`

Reply:
I've changed it to type because the name is read-only `Cannot assign to 'name' because it is a read-only property.`

Reply:
I could do it this way
```tsx
  const blitzSessionMiddleware: Middleware<{
    cookiePrefix?: string
  }> = async (req, res, next) => {
    if (req.method !== "HEAD" && !res.blitzCtx.session) {
      // This function also saves session to res.blitzCtx
      await getSessionContext(req, res)
    }
    return next()
  }
  blitzSessionMiddleware.config = {
    cookiePrefix: config.cookiePrefix,
  }
  return blitzSessionMiddleware
```
This way `name === blitzSessionMiddleware` but is it clear enough?

Reply:
Ah yeah I see. And yeah let's roll with that for now

Reply:
Should I switch it or leave the current implementation with type?